### PR TITLE
Added early exit & tests for LAPACK

### PR DESCRIPTION
**Pull request by kshyatt**
Made it so the two versions of `sytrf!` do the same thing. More tests.

### CODE REVIEW

File: base/linalg/lapack.jl [hunk 1]
```diff
@@ -3020,6 +3020,9 @@ for (sysv, sytrf, sytri, sytrs, elty, relty) in
             n = chksquare(A)
             chkuplo(uplo)
             ipiv  = similar(A, BlasInt, n)
+            if n == 0
+                return A, ipiv
```
Comment (tkelman) [inline base/linalg/lapack.jl]: what happens if `n == 0` and the `ccall` is allowed to run?

Comment (kshyatt) [inline base/linalg/lapack.jl]: Nothing bad.

**Comment by kshyatt** [comment]
Anyone got any complaints about me merging this bad boy?

**Comment by ViralBShah** [comment]
Looks good.

**Pull request merged**